### PR TITLE
Work around for Issue https://github.com/dnSpyEx/dnSpy/issues/96

### PR DESCRIPTION
**Pull request by josegomez**
This is a work-around hack for the issue Reported: https://github.com/dnSpyEx/dnSpy/issues/96

### CODE REVIEW

File: Extensions/dnSpy.Debugger/dnSpy.Debugger.DotNet.CorDebug/Impl/ModuleCreator.cs [hunk 1]
```diff
@@ -92,10 +92,18 @@ static class ModuleCreator {
...
-			if (comMetadata is null)
-				throw new InvalidOperationException();
-			var result = new DmdLazyMetadataBytesCom(comMetadata, engine.GetDynamicModuleHelper(dnModule), engine.DmdDispatcher);
-			return () => result;
+			if (comMetadata == null) {
+				if (comMetadata is null && dnModule.Process.CorProcess.CLRVersion.Major != 6) {
+					throw new InvalidOperationException();
+				}
+				else if (comMetadata is null && dnModule.Process.CorProcess.CLRVersion.Major == 6) //Net 6 hack for Issue: https://github.com/dnSpyEx/dnSpy/issues/96
+					return () => new DmdLazyMetadataBytesNull();
```
Comment (riQQ) [inline Extensions/dnSpy.Debugger/dnSpy.Debugger.DotNet.CorDebug/Impl/ModuleCreator.cs:100]: The null check is duplicated: `comMetadata == null` L95 and `comMetadata is null` in L95 and L99.
```suggestion
			if (comMetadata is null) {
				if (dnModule.Process.CorProcess.CLRVersion.Major != 6) {
					throw new InvalidOperationException();
				}
				else if (dnModule.Process.CorProcess.CLRVersion.Major == 6) //Net 6 hack for Issue: https://github.com/dnSpyEx/dnSpy/issues/96
					return () => new DmdLazyMetadataBytesNull();
```

Comment (riQQ) [inline Extensions/dnSpy.Debugger/dnSpy.Debugger.DotNet.CorDebug/Impl/ModuleCreator.cs:100]: The null check is duplicated: `comMetadata == null` L95 and `comMetadata is null` in L95 and L99.
```suggestion
			if (comMetadata is null) {
				if (dnModule.Process.CorProcess.CLRVersion.Major != 6)
					throw new InvalidOperationException();
				else //Net 6 hack for Issue: https://github.com/dnSpyEx/dnSpy/issues/96
					return () => new DmdLazyMetadataBytesNull();
```

Comment (josegomez) [reply Extensions/dnSpy.Debugger/dnSpy.Debugger.DotNet.CorDebug/Impl/ModuleCreator.cs]: Yup, Fixed. Pushing!

File: Extensions/dnSpy.Debugger/dnSpy.Debugger.DotNet.CorDebug/Impl/ModuleCreator.cs [hunk 2]
```diff
@@ -92,10 +92,18 @@ static class ModuleCreator {
...
-			if (comMetadata is null)
-				throw new InvalidOperationException();
-			var result = new DmdLazyMetadataBytesCom(comMetadata, engine.GetDynamicModuleHelper(dnModule), engine.DmdDispatcher);
-			return () => result;
+			if (comMetadata == null) {
+				if (dnModule.Process.CorProcess.CLRVersion.Major != 6) {
+					throw new InvalidOperationException();
+				}
+				else  //Net 6 hack for Issue: https://github.com/dnSpyEx/dnSpy/issues/96
+					return () => new DmdLazyMetadataBytesNull();
```
Comment (ElektroKill) [inline Extensions/dnSpy.Debugger/dnSpy.Debugger.DotNet.CorDebug/Impl/ModuleCreator.cs:100]:
```suggestion
				if (dnModule.Process.CorProcess.CLRVersion.Major == 6) {
					// Workaround for a .NET 6 bug where dynamic modules don't have a IMetaDataImport2 interface when a debugger is attached
					// see https://github.com/dnSpyEx/dnSpy/issues/96
					return () => new DmdLazyMetadataBytesNull();
				}

				throw new InvalidOperationException();
```
Reason: Inverted `if` for clarity, `else` clause removed as it was redundant. Comment updated with additional details.

**Review by ElektroKill — CHANGES_REQUESTED** [review]

File: Extensions/dnSpy.Debugger/dnSpy.Debugger.DotNet.Metadata/DmdLazyMetadataBytes.cs [hunk 3]
```diff
@@ -151,4 +151,15 @@ public sealed class DmdLazyMetadataBytesCom : DmdLazyMetadataBytes {
...
+	/// </summary>
+	public sealed class DmdLazyMetadataBytesNull : DmdLazyMetadataBytes {
+		/// <summary>
+		/// Constructor
+		/// </summary>
+		public DmdLazyMetadataBytesNull() { }
+	}
+
+
 }
```
Comment (ElektroKill) [inline Extensions/dnSpy.Debugger/dnSpy.Debugger.DotNet.Metadata/DmdLazyMetadataBytes.cs:165]:
```suggestion
}
```
Reason: These two blank lines are unnecessary.

File: Extensions/dnSpy.Debugger/dnSpy.Debugger.DotNet.Metadata/DmdLazyMetadataBytes.cs [hunk 4]
```diff
@@ -151,4 +151,15 @@ public sealed class DmdLazyMetadataBytesCom : DmdLazyMetadataBytes {
 			Dispatcher = dispatcher ?? throw new ArgumentNullException(nameof(dispatcher));
 		}
 	}
+	/// <summary>
```
Comment (ElektroKill) [inline Extensions/dnSpy.Debugger/dnSpy.Debugger.DotNet.Metadata/DmdLazyMetadataBytes.cs:154]:
```suggestion

	/// <summary>
```
Reason: New line between class definitions, consistency.

**Review by ElektroKill — CHANGES_REQUESTED** [review]
Just a few things I missed in my previous review. Otherwise LGTM!

**Review by ElektroKill — APPROVED** [review]

**Pull request merged**